Tasks/AzureRmWebAppDeployment/AzureRMUtil.ts

```diff
@@ -0,0 +1,293 @@
+
+    }
+    
+    return deferred.promise;
+};
+
+function getAzureRMWebAppDetails_version2 ( SPN, webAppName : string, resourceType : string ) : Q.Promise<any> {
+
+	var deferred = Q.defer<any>();
+
+	getAuthorizationToken(SPN)
+	.then(function (accessToken : string) {
+		
+ 		var requestOptions = {
+ 		 	url : 'https://management.azure.com/subscriptions/' + SPN.subscriptionId + '/resources?$filter=resourceType EQ \'' + resourceType + '\' AND substringof(\'' + webAppName + '\', name)&api-version=2016-07-01',
+ 		 	auth : {
+ 		 		bearer : accessToken
+ 		 	}
+ 		 };
+
+ 		request (requestOptions, function (error, response, body) {
+ 		 	if ( error ) 
+ 		 		deferred.reject (error);
+ 		 	else {
+ 		 		if ( response.statusCode === 200 ) {
+ 		 			var obj = JSON.parse (body);
+ 		 			deferred.resolve (obj.value[0]);
+ 		 		}
+ 		 		else deferred.reject ('Error Occurred : [Staus Code : '+response.statusCode+']');
+
+ 		 	}
+ 		});
+ 		
+	}, function (error) {
+		taskLib.error(error);
+	});
+
+	return deferred.promise;
+
+}
+exports.getAzureRMWebAppDetails_version2 = getAzureRMWebAppDetails_version2;
+
+function updateDeploymentStatus (azureRMWebAppConnectionDetails, deployAzureWebsiteError : boolean) {
+
+	var webAppPublishKuduUrl = azureRMWebAppConnectionDetails.KuduHostName;
+	
+	if ( webAppPublishKuduUrl ) {
+
+		var status = 3;	 // Set default status as failed 
+		var status_text = 'failed';
+
+		if ( !deployAzureWebsiteError ) {
+			status = 4;
+			status_text = 'succeeded';
+		}
+
+		/* ----------------------------------------------------------------------------------------
+		username = azureRMWebAppConnectionDetails.UserName;
+		securePwd = < convert to secure string > azureRMWebAppConnectionDetails.UserPassword;
+
+		credential = System.Management.Automation.PSCredential (username, securePwd);
+		------------------------------------------------------------------------------------------ */
+
+		// Get Author of build or release 
+
+		var author = taskLib.getVariable ('build.sourceVersionAuthor');
+		if ( author === undefined )  {
+			author = taskLib.getVariable ('build.requestedfor');
+			if ( author === undefined )
+				author = taskLib.getVariable ('release.requestedfor');
+
+			if ( author === undefined )
+				author = taskLib.getVariable ('agent.name');
+		}
+		
+		// Use buildId/releaseId to update deployment Status 
+		// Use buildUrl/releaseUrl to update deployment message 
+
+		var buildUrlTaskVar = taskLib.getVariable ('build.buildUri'), 
+			releaseUrlTaskVar = taskLib.getVariable ('release.releaseUri');
+
+		// Get the build id/number
+		var buildIdTaskVar = taskLib.getVariable ('build.buildId'), 
+			releaseIdTaskVar = taskLib.getVariable ('release.releaseId');
+
+		var collectionUrl = taskLib.getVariable ('system.TeamFoundationCollectionUri'), 
+			teamProject = taskLib.getVariable ('system.teamProject');	
+
+		var buildOrReleaseUrl = "" ;
+		var uniqueId = Date.now();		// //uniqueId = Get current date ddMMyyhhmmss ;
```

Comment:
i suggest put the logic of getting a unique ID in a small method.
var uniqueId = GetUniqueIdForDeployment();
function GetUniqueIdForDeployment() : string {
return Date.now();
}

This way - the basic decisions made by the task are easily locatable.